pkg/resourcecreator/deployment.go

```diff
@@ -348,3 +413,14 @@ func leaderElectionContainer(name, namespace string) corev1.Container {
 		Args: []string{fmt.Sprintf("--election=%s", name), "--http=localhost:4040", fmt.Sprintf("--election-namespace=%s", namespace)},
 	}
 }
+
+func GetContainerByName(containersRef *[]corev1.Container, name string) *corev1.Container {
```

Comment:
Hvorfor endre denne til `containersRef *[]corev1.Container` ?

Reply:
Den måtte endres da den første iterasjonen muterte objektet i stedet for å kopiere og returnere et nytt. Endret dette fordi det ikke ble like lesbart, og resten av tilsvarende caser ble løst med DeepCopy

Reply:
byttes tilbake, da det ikke er en kopi av hele objektet som sendes så vi kan fremdeles returnere en peker til det opprinnelige objektet